### PR TITLE
Calendar week numbers

### LINKED ISSUE #6172: View week numbers in datepicker and calendar
## Is your feature request related to a problem? Please describe.
Our customers need to see the week number when they pick any date in our software. The current igx controls datepicker and calendar have no option for that.

## Describe the solution you'd like
I'd like to have an option for activating week numbers in the datepicker and calendar controls. Something like [showWeekNumber]="true"

## Describe alternatives you've considered
-

## Additional context
-

### PR DESCRIPTION
**Pull request by Aleksandyr**
Closes #6172 

### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Comment by Aleksandyr** [comment]
Remains:
- [ ] Unit tests
- [ ] Design

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 1]
```diff
@@ -574,18 +592,122 @@
         display: flex;
         justify-content: space-between;
         margin: $cal-row-margin;
-        padding: $cal-row-padding;
+        padding: 0;
 
         %cal-value {
             &:first-of-type {
-                padding-left: em(8px);
+                padding-left: $cal-row-padding;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:599]: This should use the interpolated $left variable for RTL compatibility.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 2]
```diff
@@ -574,18 +592,122 @@
...
             }
 
             &:last-of-type {
-                padding-right: em(8px);
+                padding-right: $cal-row-padding;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:603]: Same here.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 3]
```diff
@@ -574,18 +592,122 @@
...
+                min-width: 12.5% !important;
             }
         }
 
+        %cal-value-date--week-number {
+            padding-#{$right}: 0 !important;
+        }
+
+        %cal-value-date--week-number + %cal-value {
+            padding-left: 0 !important;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:623]: Again, cover RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 4]
```diff
@@ -574,18 +592,122 @@
...
+            &:after {
+                display: none;
+            }
+
+            &:before {
+                content: '';
+                position: absolute;
+                width: 100%;
+                height: $cal-row-padding / 2;
+                left: 0;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:660]: RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 5]
```diff
@@ -574,18 +592,122 @@
...
+                top: 100%;
+                background: --var($theme, 'week-number-background') !important;
+            }
+        }
+
+        %label--week-number {
+            padding-right: 0 !important;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:667]: RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 6]
```diff
@@ -574,18 +592,122 @@
...
+                &:after {
+                    content: '';
+                    position: absolute;
+                    width: 100%;
+                    background: --var($theme, 'week-number-background');
+                }
+
+                &:after {
+                    top: 100%;
+                    left: 0;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:685]: RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 7]
```diff
@@ -574,18 +592,122 @@
...
+                    @if $not-bootstrap-theme{
+                        height: ($cal-row-padding / 2) + rem(2px);
+                    } @else {
+                        height: $cal-row-padding + rem(5px);
+                    }
+                }
+
+                &:before {
+                    bottom: 100%;
+                    left: 0;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:695]: RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 8]
```diff
@@ -574,18 +592,122 @@
...
+                    height: $cal-row-padding;
+                    border-top-left-radius: $border-radius;
+                    border-top-right-radius: $border-radius;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:698]: RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 9]
```diff
@@ -574,18 +592,122 @@
...
+                }
+            }
+        }
+
+        %label--week-number + %cal-value--label {
+            padding-left: 0 !important;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:704]: RTL.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss [hunk 10]
```diff
@@ -594,6 +716,31 @@
...
+        &:nth-child(2) {
+            @if $not-bootstrap-theme{
+                margin-top: 0;
+            }
+        }
+
+        &:last-of-type {
+            %cal-value-content--week-number {
+                border-bottom-left-radius: $border-radius;
+                border-bottom-right-radius: $border-radius;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:730]: RTL.

**Review by simeonoff — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss (hunk 8 above)
Comment (desig9stein) [reply projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:698]: This is not affected by RTL. The radius is the same for all corners.

File: projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss (hunk 10 above)
Comment (desig9stein) [reply projects/igniteui-angular/src/lib/core/styles/components/calendar/_calendar-theme.scss:730]: This is not affected by RTL. The radius is the same for all corners.

**Review by simeonoff — APPROVED** [review]

**Review by zdrawku — APPROVED** [review]

**Pull request merged**